### PR TITLE
Convert remaining KnownMatrix children to using _unitary_

### PR DESCRIPTION
**Pull request by Strilanc**

### CODE REVIEW

File: cirq/google/xmon_gates.py [hunk 1]
```diff
@@ -213,18 +212,11 @@ def to_proto_dict(self, *qubits):
...
-        return super().try_cast_to(desired_type, ext)
-
-    def has_matrix(self):
-        return not isinstance(self.half_turns, value.Symbol)
-
-    def matrix(self):
-        if not self.has_matrix():
-            raise ValueError("Don't have a known matrix.")
-        return protocols.unitary(ops.Rot11Gate(half_turns=self.half_turns))
+    def _unitary_(self) -> Optional[np.ndarray]:
```
Comment (maffoo) [inline cirq/google/xmon_gates.py]: I thought the plan was to change the return type here to `Union[np.ndarray, type(NotImplemented)]`

File: cirq/google/xmon_gates.py [hunk 2]
```diff
@@ -542,13 +527,11 @@ def inverse(self):
...
-    def has_matrix(self):
-        return not isinstance(self.half_turns, value.Symbol)
-
-    def matrix(self):
-        if not self.has_matrix():
-            raise ValueError("Don't have a known matrix.")
-        return np.diag([(-1j)**self.half_turns, 1j**self.half_turns])
+    def _unitary_(self) -> Optional[np.ndarray]:
+        if isinstance(self.half_turns, value.Symbol):
+            return None
```
Comment (maffoo) [inline cirq/google/xmon_gates.py]: Why `None` here instead of `NotImplemented` like the methods above?

File: cirq/google/xmon_gates.py (hunk 1 above)
Comment (Strilanc) [reply cirq/google/xmon_gates.py]: Good catch. Done.

File: cirq/google/xmon_gates.py (hunk 2 above)
Comment (Strilanc) [reply cirq/google/xmon_gates.py]: Good catch. Done.

**Comment by Strilanc** [comment]
@maffoo @dabacon take another look

File: cirq/circuits/convert_to_cz_and_single_gates_test.py [hunk 3]
```diff
@@ -20,15 +20,15 @@
 
 
 def test_avoids_infinite_cycle_when_matrix_available():
-    class OtherX(cirq.Gate, cirq.KnownMatrix, cirq.CompositeGate):
-        def matrix(self):
+    class OtherX(cirq.Gate, cirq.CompositeGate):
+        def _unitary_(self):
```
Comment (dabacon) [inline cirq/circuits/convert_to_cz_and_single_gates_test.py]: return type?  Here an elsewhere in this pull request.

**Pull request merged**